### PR TITLE
[JENKINS-35459] - Prevent Duplicated elements with incorrect URL when using the search on DashBoard View

### PR DESCRIPTION
**Pull request by alexanderrtaylor**
Fixing the search bar not working for folders because Felix’s PR is
good: https://github.com/jenkinsci/jenkins/pull/2401 but needed to fix
some tests

https://issues.jenkins-ci.org/browse/JENKINS-35459

### Changelog entry

* JENKINS-35459, Bug: Prevent Duplicated elements with incorrect URL when using the search on DashBoard View

### CODE REVIEW

File: core/src/main/java/hudson/model/ListView.java
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/ListView.java]: Breaks formatting 🐛

File: core/src/main/java/hudson/model/ListView.java
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/ListView.java]: breaks correct formatting

File: core/src/main/java/hudson/model/ListView.java
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/ListView.java]: `@param` should be lowercase

File: core/src/main/java/hudson/model/ListView.java
Comment (oleg-nenashev) [inline core/src/main/java/hudson/model/ListView.java]: Why is there `false` here?

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
There are breaking formatting changes. If you are modifying the existing commits, you also need to keep the reference to the contributor in the commit history or to get his permission to not reference him. CC @fbelzunc

File: core/src/main/java/hudson/model/ListView.java
Comment (alexanderrtaylor) [reply core/src/main/java/hudson/model/ListView.java]: Actually that is a good suggestion. I would think you want to recursively search over getItems. I am changing it to true and testing to see if that is behavior we want

**Review by oleg-nenashev — APPROVED** [review]
🐝 , just minor comments mentioned above

**Comment by aheritier** [comment]
LGTM 🐝

**Review by aheritier — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Needs feedback from @daniel-beck . Keeping on-hold

File: test/src/test/java/hudson/search/SearchTest.java
Comment (daniel-beck) [inline test/src/test/java/hudson/search/SearchTest.java]: leftover debug code?

**Review by daniel-beck — COMMENTED** [review]
Would be useful if comments (can be PR comments, not code comments) indicated how the tests would fail before this fix.

File: test/src/test/java/hudson/search/SearchTest.java
Comment (oleg-nenashev) [reply test/src/test/java/hudson/search/SearchTest.java]: Right, it should be Logger at least. Otherwise streams won't be syncronized

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
🐛 as @daniel-beck noticed, logging in tests needs to be cleaned up

**Review by oleg-nenashev — COMMENTED** [review]
@alexanderrtaylor Was there any specific reason to remove Javadoc of the public method?

**Comment by alexanderrtaylor** [comment]
@oleg-nenashev 

You had commented: 
"oleg-nenashev 19 days ago Owner
Fine to remove custom Javadoc from what I see"

I thought that is what you were referring to but maybe I missed what you were saying?

**Review by oleg-nenashev — APPROVED** [review]
Or I just have no memory of this place...
Hence 🐝

**Comment by oleg-nenashev** [comment]
@revuewbybees done

File: core/src/main/java/hudson/model/ListView.java
Comment (daniel-beck) [inline core/src/main/java/hudson/model/ListView.java]: What is this for? Does this need `@Override`, `@Restricted` or `@since`?

Comment (daniel-beck) [reply core/src/main/java/hudson/model/ListView.java]: Yep, lost `@Override` without explanation.

**Comment by oleg-nenashev** [comment]
@daniel-beck do you require `@Override` to be restored?

**Comment by oleg-nenashev** [comment]
It has 2 votes, so it's ready for merge.

**Comment by daniel-beck** [comment]
@oleg-nenashev No strong opinion, but diff looked suspiciously like something newly added without restriction or doc.

**Comment by daniel-beck** [comment]
What happened to

> If you are modifying the existing commits, you also need to keep the reference to the contributor in the commit history or to get his permission to not reference him. CC @fbelzunc

**Comment by oleg-nenashev** [comment]
@alexanderrtaylor @fbelzunc ping regarding the contribution references. We will not merge the PR until there is a fix or approval

**Comment by fbelzunc** [comment]
@oleg-nenashev @alexanderrtaylor I don't care about my previous commits :-) You can merge it!

**Comment by oleg-nenashev** [comment]
@fbelzunc thanks for the response!

File: core/src/main/java/hudson/model/ListView.java
Comment (oleg-nenashev) [reply core/src/main/java/hudson/model/ListView.java]: I will cleanup the Javadoc / Override changes after the merge

**Pull request merged**
